rs/nns/governance/src/governance.rs

```diff
@@ -1803,6 +1803,30 @@ fn spawn_in_canister_env(future: impl Future<Output = ()> + Sized + 'static) {
 }
 
 impl Governance {
+    /// Creates a new Governance instance with uninitialized fields. The canister should only have
+    /// such state before the state is recovered from the stable memory in post_upgrade or
+    /// initialized in init. In any other case, the `Governance` object should be initialized with
+    /// either `new` or `new_restored`.
+    pub fn new_uninitialized(
+        env: Box<dyn Environment>,
+        ledger: Box<dyn IcpLedger>,
+        cmc: Box<dyn CMC>,
```

Comment:
NCR. Why have > 0 parameters?

Reply:
The "production code version" of them are in `canister.rs`. I don't see a compelling reason to change that pattern yet. I think it would be possible to initialize them with some dummy impl of those traits, but it doesn't seem to be worth the effort or the complexity.